controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -269,6 +290,42 @@ func (in *KubeadmControlPlane) validateCommon() (allErrs field.ErrorList) {
 		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "version"), in.Spec.Version, "must be a valid semantic version"))
 	}
 
+	if in.Spec.RolloutStrategy != nil {
```

Comment:
Should we error out in case in.Spec.RolloutStrategy == nil?

Reply:
Can `if in.Spec.RolloutStrategy == nil` ever be true? I assumed that defaulting on line 68 makes sure that the value is never nil. Or have I misunderstood?

Reply:
Given that the defaulting function could change at some point, I personally prefer to have each func self consistent (or at least to explicitly document the assumption each function relies on, especially if those assumptions depends on something somewhere else in the codebase), but this is just a nit, feel free to ignore it

Reply:
Thanks for the clarification @fabriziopandini.